src/blurredLocationDisplay.js

```diff
@@ -72,83 +96,160 @@ BlurredLocationDisplay = function BlurredLocationDisplay(options) {
+                  title: title
+            }) ;
+            if(filterCoordinate(latitude , longitude)){
+              afterDecimal = latitude.toString().split(".")[1] ;
+              precision = 0 ; 
+              if(typeof afterDecimal !== "undefined") {
+                precision = afterDecimal.length ;
+              }
+              m.addTo(map).bindPopup("<a href=" + url + ">" + title + "</a> <br> Precision : " + precision) ;
+              PLmarkers_array[PLmarkers_array.length] = m ;  
+            }
+
+          }
+        }
+      });
+    }
+  }
+
+  function fetchLocationData(isOn) {
+    if(isOn === true) {
+      for(i=0 ; i < options.locations.length ; i++){
+        var latitude = options.locations[i][0] ; 
+        var longitude = options.locations[i][1] ; 
+        var BlackIcon = new L.Icon.BlackIcon() ;
+        if(filterCoordinate(latitude , longitude)){
+              afterDecimal = latitude.toString().split(".")[1] ;
+              precision = 0 ; 
+              if(typeof afterDecimal !== "undefined") {
+                precision = afterDecimal.length ;
+              }
+              var m = L.marker([latitude, longitude] , {icon: BlackIcon}) ;
+              m.addTo(map).bindPopup("Precision : " + precision) ;
+              locations_markers_array[locations_markers_array.length] = m ;
+        }
+      } 
     }
+  }
 
+  function fetchSourceUrlData(isOn) {
+    if(isOn === true) {
+      var NWlat = map.getBounds().getNorthWest().lat ;
+      var NWlng = map.getBounds().getNorthWest().lng ;
+      var SElat = map.getBounds().getSouthEast().lat ;
+      var SElng = map.getBounds().getSouthEast().lng ;
+
+      source_url = options.source_url + "?nwlat=" + NWlat + "&selat=" + SElat + "&nwlng=" + NWlng + "&selng=" + SElng ; 
+      $.getJSON(source_url , function (data) {
+        if (!!data.items) {
+          for (i = 0; i < data.items.length; i++) {
+            var RedIcon = new L.Icon.RedIcon() ;
+            var mid = data.items[i].doc_id ;
+            var url = data.items[i].doc_url;
+            var latitude = data.items[i].latitude ;
+            var longitude = data.items[i].longitude ;
+            var title = data.items[i].doc_title;
+            var m = L.marker([data.items[i].latitude, data.items[i].longitude], {
+                  icon: RedIcon
+            }) ;
+            if(filterCoordinate(latitude , longitude)){
+              afterDecimal = latitude.toString().split(".")[1] ;
+              precision = 0 ; 
+              if(typeof afterDecimal !== "undefined") {
+                precision = afterDecimal.length ;
+              }
+              m.addTo(map).bindPopup("<a href=" + url + ">" + title + "</a> <br> Precision : " + precision) ;
+              SourceUrl_markers_array[SourceUrl_markers_array.length] = m ;  
+            }
+          }
+        }
+      });  
+    }
+  }
+
+  function fetchPLpeopleAPI() {
     map.on('zoomend' , function () {
```

Comment:
If we run all remote data sources through the same standard handlers, these would become generic.

Also, shouldn't we just be:

1. listening for map interactions
2 triggering fetch from any `source_url`
3. waiting for response
4. storing all, but filtering to decide which are displayed?

or something like that?

Reply:
Yes @jywarren ...will make code the generic 👍  .

We are doing that only : 
1.)    listening for map interactions
https://github.com/sagarpreet-chadha/leaflet-blurred-location-display/blob/ae7806f6089de011473b7ac0998a6f84d8d4ddb9/src/blurredLocationDisplay.js#L205

2.) triggering fetch from any `source_url` and waiting for response
https://github.com/sagarpreet-chadha/leaflet-blurred-location-display/blob/ae7806f6089de011473b7ac0998a6f84d8d4ddb9/src/blurredLocationDisplay.js#L177

3.)  storing all, but filtering to decide which are displayed?

https://github.com/sagarpreet-chadha/leaflet-blurred-location-display/blob/ae7806f6089de011473b7ac0998a6f84d8d4ddb9/src/blurredLocationDisplay.js#L189

AND

https://github.com/sagarpreet-chadha/leaflet-blurred-location-display/blob/ae7806f6089de011473b7ac0998a6f84d8d4ddb9/src/blurredLocationDisplay.js#L66

What do you think ?

Reply:
Awesome I assumed you did this but I think the generic version of the code will make it easier to see this. Let's do it and review again! Thanks Sagarpreet!